lighthouse-core/gather/gather-runner.js

```diff
@@ -270,6 +278,12 @@ class GatherRunner {
           });
         });
         return artifacts;
+      })
+      // cleanup on error
+      .catch(err => {
+        GatherRunner.disposeDriver(driver);
```

Comment:
nice job adding this here. We should have been doing this for other errors already